src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs

```diff
@@ -148,6 +157,38 @@ public async Task<bool> IsAdminAsync(Guid? userId = null)
       }
 
       return await IsUserAdminAsync(userId.Value);
+    }
+
+    public async Task<bool> IsManagerAsync(ManagerSource managerSource, Guid entityId)
+    {
+      Guid userId = _httpContext.GetUserId();
+
+      Response<IOperationResult<bool>> result = null;
+
+      switch (managerSource)
+      {
+        case ManagerSource.Project:
+          result = await _requestClientPM.GetResponse<IOperationResult<bool>>(
+            ICheckProjectManagerRequest.CreateObj(userId, entityId),
+            timeout: 5000);
+          break;
+        case ManagerSource.Department:
+          result = await _requestClientDM.GetResponse<IOperationResult<bool>>(
+            ICheckDepartmentManagerRequest.CreateObj(userId, entityId),
+            timeout: 5000);
+          break;
+        default: 
+          return false;
+      }
+
+      if (!result.IsSuccess())
+      {
+        _logger.LogWarning("Failed to send request to CheckEntityManager via the broker.");
```

Comment:
May be add to message source (Project/Department)?

Reply:
done